src/Raven.Server/Documents/Handlers/Debugging/QueriesDebugHandler.cs

```diff
@@ -93,5 +95,112 @@ public Task RunningQueries()
+                        {
+                            curDjvItem["IsDistinct"] = true;                                                        
+                        }
+
+                        if (item.HasFacet)
+                        {
+                            curDjvItem["HasFacet"] = true;                            
+                        }
+
+                        if (item.HasMoreLikeThis)
+                        {
+                            curDjvItem["HasMoreLikeThis"] = true;                            
+                        }
+
+                        if (item.HasSuggest)
+                        {
+                            curDjvItem["HasSuggest"] = true;                            
+                        }
+
+                        if (item.OrderBy != null)
+                        {
+                            curDjvItem["OrderBy"] = true;                            
+                        }
+
+                        if (item.HasCmpXchg)
+                        {
+                            curDjvItem["HasCmpXchg"] = true;
+                        }
+
+                        if (item.HasExplanations)
+                        {
+                            curDjvItem["HasExplanations"] = true;
+                        }
+
+                        if (item.HasIntersect)
+                        {
+                            curDjvItem["HasIntersect"] = true;
+                        }
+
+                        if (item.IsDynamic)
+                        {
+                            curDjvItem["IsDynamic"] = true;
+                        }
+
+                        if (item.IsOptimizedSortOnly)
+                        {
+                            curDjvItem["IsOptimizedSortOnly"] = true;
+                        }        
+
+                        if (item.SelectFields.Any(x => x.Function != null))
+                        {
+                            curDjvItem["IsJSProjection"] = true;
+                        }
+                        
+                        if (string.IsNullOrEmpty(item.CollectionName) == false)
+                            curDjvItem["CollectionName"] = item.CollectionName;                                                
+
+                        if (string.IsNullOrEmpty(item.AutoIndexName) == false)
+                        {
+                            curDjvItem["AutoIndexName"] = item.AutoIndexName;                                
+                        }
+
+                        if (string.IsNullOrEmpty(item.IndexName) == false)
+                        {
+                            curDjvItem["IndexName"] = item.IndexName;
+                        }                        
+                    
+                        curDjvItem["QueryText"] = item.QueryText;                        
+                    }
+                }
+
+                var blittableResult = context.ReadObject(responseDjv, "debug/queries/cache/list");
```

Comment:
IMO we are missing some stuff:
1. We should add CreatedAt
2. We should add LastQueriedAt
3. We need to add total number of queries 
4. We need to add current number of queries

Reply:
I agree about 3,4 . I've discussed with @ayende similiar issue question to 1,2 yesterday and I'm not sure how relevant this will be for the ever-changing nature of the cache (512 items etc)

Reply:
What is 3 & 4? I get total number of queries, but what is current number of queries?
1) Is easy, 2) will be best effort.

Reply:
I wouldn't say ever changing, I know that they can overlap but in decent amount of cases cache will (more or less) stabilize after few minutes IMO.

Reply:
@ayende 
Total is = 'how many times we took this Query from cache' 
and 
Current is = 'how many queries for that particular Query are done at the moment'

Reply:
This could help us to know which ones are actively used.

Reply:
Current probably means `Interlocked.Increment` all over the place. 
And will likely create hot spot for concurrency.

Reply:
@ppekrol This will be confusing. You don't want to use the cache as metrics, because queries go in and out, so the data will not be complete

Reply:
@ayende I'm not sure if Interlocked.Increment would be a big problem here? But since current is not a big deal here, then we can drop it. But Total would be good (even if we know that this might be dropped from cache).

@maximburyak as I said, I think that most of the time the cache will stabilize. I would not use this as a metric. This gives you cache state and traffic.